src/include/migraphx/migraphx_float8.hpp

```diff
@@ -0,0 +1,387 @@
+};
+
+template <>
+class numeric_limits<fp8e5m2>
+{
+    public:
+    static constexpr fp8e5m2 epsilon() { return fp8e5m2(0x34, fp8e5m2::from_bits()); }
+    // 7D, 7E, 7F are positive NaNs and FD, FE, FF are negative NaNs
+    static constexpr fp8e5m2 quiet_NaN() { return fp8e5m2(0xFF, fp8e5m2::from_bits()); }
+
+    static constexpr fp8e5m2 max() { return fp8e5m2(0x7B, fp8e5m2::from_bits()); }
+    // this is min value that is not DeNorm. DeNorm min is 0x01. I am not sure if we want to make
+    // this distinction. For the floating points we would end up using lowest most of the times.
+    static constexpr fp8e5m2 min() { return fp8e5m2(0x4, fp8e5m2::from_bits()); }
+
+    static constexpr fp8e5m2 lowest() { return fp8e5m2(0xFB, fp8e5m2::from_bits()); }
+    // 7C and FC both are infinity
+    static constexpr fp8e5m2 infinity() { return fp8e5m2(0x7C, fp8e5m2::from_bits()); }
+};
+} // namespace migraphx_fp8
+
+// =================================================================================================
+// define numeric limits for the new data type
+namespace std {
+
+#define MIGRAPHX_FP8_STD_OVERLOADS(T)                                \
+    inline bool isfinite(T x) { return x.is_inf(); }                 \
+    inline bool isnan(T x) { return x.is_nan(); }                    \
+    template <>                                                      \
+    class numeric_limits<T> : public migraphx_fp8::numeric_limits<T> \
+    {                                                                \
+    };                                                               \
+    template <class U>                                               \
+    struct common_type<T, U> : std::common_type<float, U>            \
+    {                                                                \
+    };                                                               \
+    template <class U>                                               \
+    struct common_type<U, T> : std::common_type<float, U>            \
+    {                                                                \
+    };                                                               \
+    template <>                                                      \
+    struct common_type<T, T>                                         \
+    {                                                                \
+        using type = T;                                              \
+    };
+
+MIGRAPHX_FP8_STD_OVERLOADS(migraphx_fp8::fp8e4m3fn)
+MIGRAPHX_FP8_STD_OVERLOADS(migraphx_fp8::fp8e5m2)
+MIGRAPHX_FP8_STD_OVERLOADS(migraphx_fp8::fp8e4m3fnuz)
+MIGRAPHX_FP8_STD_OVERLOADS(migraphx_fp8::fp8e5m2fnuz)
+
+} // namespace std
+// =================================================================================================
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
+#endif // MIGRAPHX_GUARD_RTGLIB_FLOAT8_HPP
```

Comment:
Also, this header should be named just `float8.hpp`. It doesn't make sense to prefix it with migraphx when it already is in the migraphx directory.

Reply:
Done